agent/structs/structs.go

```diff
@@ -765,6 +765,57 @@ type ServiceConnect struct {
 	SidecarService *ServiceDefinition `json:",omitempty" bexpr:"-"`
 }
 
+// Merge overlays the given node's attributes onto the existing node.
```

Comment:
Not super clear from the comment which service's attributes take precedence. I assumed initially it was the reverse of the implementation. Maybe we could state explicitly that `other` node's properties will take precedence if they are not set as well as mention the recursive behaviour for tags, checks, config maps etc.?